engine/gamesys/src/gamesys/components/comp_gui.cpp

```diff
@@ -1135,92 +1135,105 @@ namespace dmGameSystem
-
-                    const float* point = &points[i * 2];
-                    const float* uv = &uvs[i * 2];
-                    // COnvert from range [-0.5,+0.5] to [0.0, 1.0]
-                    float x = point[0] * scaleX + 0.5f;
-                    float y = point[1] * scaleY + 0.5f;
-
-                    Vector4 p = w * Point3(x, y, 0.0f);
-                    BoxVertex v(p, uv[0], uv[1], pm_color);
-                    gui_world->m_ClientVertexBuffer.Push(v);
+                    // Since we don't use an index buffer, we duplicate the vertices manually
+                    uint32_t index_count = geometry->m_Indices.m_Count;
+                    for (uint32_t index = 0; index < index_count; ++index)
+                    {
+                        uint32_t i = geometry->m_Indices.m_Data[index];
+                        i = reverse ? (num_points - i - 1) : i;
+
+                        const float* point = &points[i * 2];
+                        const float* uv = &uvs[i * 2];
+                        // COnvert from range [-0.5,+0.5] to [0.0, 1.0]
+                        float x = point[0] * scaleX + 0.5f;
+                        float y = point[1] * scaleY + 0.5f;
+
+                        Vector4 p = w * Point3(x, y, 0.0f);
+                        BoxVertex v(p, uv[0], uv[1], pm_color);
+                        gui_world->m_ClientVertexBuffer.Push(v);
+                    }
+
+                    rendered_vert_count += index_count;
+                    continue;
                 }
+                else if(!tc)
+                {
+                    BoxVertex v00;
+                    v00.SetColor(pm_color);
+                    v00.SetPosition(node_transforms[i] * Vectormath::Aos::Point3(0, 0, 0));
+                    v00.SetUV(0, 0);
+
+                    BoxVertex v10;
+                    v10.SetColor(pm_color);
+                    v10.SetPosition(node_transforms[i] * Vectormath::Aos::Point3(1, 0, 0));
+                    v10.SetUV(1, 0);
+
+                    BoxVertex v01;
+                    v01.SetColor(pm_color);
+                    v01.SetPosition(node_transforms[i] * Vectormath::Aos::Point3(0, 1, 0));
+                    v01.SetUV(0, 1);
+
+                    BoxVertex v11;
+                    v11.SetColor(pm_color);
+                    v11.SetPosition(node_transforms[i] * Vectormath::Aos::Point3(1, 1, 0));
+                    v11.SetUV(1, 1);
 
-                rendered_vert_count += index_count;
-                continue;
+                    gui_world->m_ClientVertexBuffer.Push(v00);
+                    gui_world->m_ClientVertexBuffer.Push(v10);
+                    gui_world->m_ClientVertexBuffer.Push(v11);
+                    gui_world->m_ClientVertexBuffer.Push(v00);
+                    gui_world->m_ClientVertexBuffer.Push(v11);
+                    gui_world->m_ClientVertexBuffer.Push(v01);
+
+                    rendered_vert_count += 6;
+                    continue;
```

Comment:
Right now, there's two checks for "if (!tc)" (also on line 1286, the else statement), which is added redundancy.
Since you collect the tc pointer first, do this check there, and continue the loop.

Reply:
Answered: https://github.com/defold/defold/pull/4921#issuecomment-644750126